sklearn/utils/validation.py

```diff
@@ -381,7 +386,10 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         array = _ensure_sparse_format(array, accept_sparse, dtype, copy,
                                       force_all_finite)
     else:
-        array = np.array(array, dtype=dtype, order=order, copy=copy)
+        # Do not physically copy memory map : if type(array) == np.memmap:
+        # type(array) == np.array
```

Comment:
The class is `numpy.ndarray`, never `numpy.array`. Also `numpy.memmap` is a sublcass of `numpy.ndarray`.